src/Compilers/Core/Portable/Syntax/InternalSyntax/SyntaxListBuilder.cs

```diff
@@ -24,6 +25,7 @@ public static SyntaxListBuilder Create()
 
         public void Clear()
         {
+            Array.Clear(_nodes, 0, Count);
```

Comment:
>Array.Clear(_nodes, 0, Count); [](http://example.com/codeflow?start=12&length=30)

Why not clear the entire array? It looks like the indexer below doesn't check/adjust the ```Count```